gdx/src/com/badlogic/gdx/utils/QuadTreeFloat.java

```diff
@@ -156,6 +157,29 @@ private void query (float centerX, float centerY, float radiusSqr, float rectX,
 		}
 	}
 
+	/** @param results For each entry found within the rectangle, if any, the value, x, and y of the entry are added to this array.
+	 *           See {@link #VALUE}, {@link #X}, {@link #Y}, and {@link #DISTSQR}. */
+	public void query (Rectangle rect, FloatArray results) {
+		if (!(x < rect.x + rect.height && x + width > rect.x && y < rect.y + rect.height && y + height > rect.y)) return;
```

Comment:
Can it be replaced by `Rectangle.contains(x,y)`?

Reply:
> Can it be replaced by `Rectangle.contains(x,y)`?

No, it would need to be replaced with Rectangle.overlaps(Rectangle). To use the check we'd have to change the constructor for the quadtree or create a new rectangle each time we query. As for the unit tests, there are already 2 tests for the quadtree in the main testing package, would it not be better to change the existing query test to include coverage for the rectangle query as well as the radius query?

Reply:
Yeah, had a better look. Instead of `x < rect.x + rect.**height**` it should be `x < rect.x + rect.**width**` right?

We can extend the current "integration test" (better than no test at all) but, imo, math classes should be covered by unit tests.

Reply:
> Yeah, had a better look. Instead of `x < rect.x + rect.**height**` it should be `x < rect.x + rect.**width**` right?
I've committed a change for that. Thanks for finding it.